sktime/utils/_testing/estimator_checks.py

```diff
@@ -97,6 +98,96 @@ def yield_estimator_checks(exclude=None):
         yield check
 
 
+def tuple_equals(x, y):
+    """Tests two tuples for equality.
+
+    Correct if tuples contain != compatible native types,
+        or pd.Series, pd.DataFrame, np.ndarray
+
+    Parameters
+    ----------
+    x: tuple
+    y: tuple
+
+    Returns
+    -------
+    bool - True if x and y have equal keys and values
+    """
+
+    n = len(x)
+
+    if n != len(y):
+        return False
+
+    # we now know dicts are same length
+    for i in range(n):
+        xi = x[i]
+        yi = y[i]
+
+        if type(xi) != type(yi):
+            return False
+
+        # we now know all types are the same
+        # so now we compare values
+        if type(xi) in [pd.DataFrame, pd.Series]:
+            if not xi.equals(yi):
+                return False
+        elif type(xi) is np.ndarray:
+            if xi.dtype != yi.dtype:
+                return False
+            if not np.array_equal(x, y, equal_nan=True):
+                return False
+        elif xi != yi:
+            return False
+
+    return True
+
+
+def dict_equals(x, y):
```

Comment:
I think this can go

Reply:
well, currently not used, but we may need it for kwargs at some point, so I wouldn't want to keep it in a text file until then

Reply:
Let's not add functionality to the main source code which isn't used, I have a separate folder for code that's related to sktime but currently not used, or perhaps add it to `contrib`?

Reply:
We might like to use it for new tests in the near future, e.g., kwargs related tests (same as current but use kwargs) or tag related tests (e.g., does it have side effects on `_tags`?) so I'd prefer to leave it in. It doesn't hurt anyone there, and it wouldn't be the precedent (see e.g., `_get_name` in the same folder). If it's really a blocker for you, I'm happy to delete it and add it back later, but it will be more difficult to find again.

Reply:
where is your separate folder if I were to put it there?

Reply:
Yes please remove it, if `_get_name` isn't used we can remove it too. My folder is just local, for smaller things I posted them on gists or anywhere else really but keeping them in the main code base isn't a good idea.

Reply:
I'm confused, @mloning - you tell me to move things to a separate folder, but that folder does not exist or is local on your computer (?).

So, would you mind making the changes yourself? It sounds like it's just a small thing and it's faster if you do it. I'm happy with a complete delete too.

Reply:
Yes just delete it! Sorry for the confusion.

Reply:
resolved below since superseded